components/elm/src/biogeophys/SoilTemperatureMod.F90

```diff
@@ -943,14 +948,49 @@ subroutine SoilThermProp (bounds,  num_nolakec, filter_nolakec, &
                   endif
                endif
             endif
+            
+            if (use_T_rho_dependent_snowthk) then ! chose which snow thermal conductivity to use 
+               if (snl(c)+1 < 1 .AND. (j >= snl(c)+1) .AND. (j <= 0)) then
+                    bw(c,j) = (h2osoi_ice(c,j)+h2osoi_liq(c,j))/(frac_sno(c)*dz(c,j))
+
+                       do i = 1, 5
+                        if (i == 1) then
+                            k_snw_vals(i) = 2.564 * (bw(c,j)/ rho_ice)**2 - 0.059 * (bw(c,j)/ rho_ice) + 0.0205
+                        else if (i == 2) then
+                            k_snw_vals(i) = 2.172 * (bw(c,j)/ rho_ice)**2 + 0.015 * (bw(c,j)/ rho_ice) + 0.0252
+                        else if (i == 3) then
+                            k_snw_vals(i) = 1.985 * (bw(c,j)/ rho_ice)**2 + 0.073 * (bw(c,j)/ rho_ice) + 0.0336
+                        else if (i == 4) then
+                            k_snw_vals(i) = 1.883 * (bw(c,j)/ rho_ice)**2 + 0.107 * (bw(c,j)/ rho_ice) + 0.0386
+                        else if (i == 5) then
+                            k_snw_vals(i) = 1.776 * (bw(c,j)/ rho_ice)**2 + 0.147 * (bw(c,j)/ rho_ice) + 0.0455
```

Comment:
Could you please add `_r8` to constants e.g. `2.564_r8 * (bw(c,j)/ rho_ice)**2 - 0.059_r8 * (bw(c,j)/ rho_ice) + 0.0205_r8`?